tests/components/test-class-component.php

```diff
@@ -512,7 +521,7 @@ public function test_set_component_theme() {
 			]
 		);
 
-		$this->assertEquals( $expected, $legacy_component->to_array()['config'] );
```

Comment:
These changes make me pause. The reason we were testing the output after running the `to_array()` method is because that method does some final tweaking too our config shape before returning a serialized JSON response. If the tests aren't passing without this change, then we probably still have a bug in our serialization step.

One thing that I think of is that during this method, the `theme_name` config property gets set from the top-level `theme` property, i.e., `$this->get_theme()`. If someone has modified the theme config property in a callback, this wouldn't be reflected in the output and we'll likely end up with a config object that looks something like this:

```js
"config": {
	"context": "site",
	"className": "",
	"style": [],
	"theme": "changedValue" // Altered in config.
	"themeName": "default", // Created during the to_array() method. Used by Irving.
	"themeOptions": [
		"default"
	]
},
```